### DISCUSSION TITLE
Singleton rule per session ?

### DISCUSSION THREAD

**Discussion by GianLucaBagnoli** [NRules/NRules#352] [Q&A] [answered]
Hello, 
	In my project I need each rule to be uniquely instantiated ( singleton ) per session.
Currently, if a given type is loaded many times into a repository then many instances of that rule are created in the same session.

In this example, 5 istances of class DummyRule1 are created:

`public class DummyRule1 : Rule { .... }

var repo = new RuleRepository();

var typeRule1 = typeof(DummyRule1);

repo.Load((loader) => loader.From(typeRule1, typeRule1, typeRule1, typeRule1, typeRule1));

var session = repo.Compile().CreateSession();`

Is there a way to enforce that constraint or it's just a responsibility of who fills the rulerepository ?

Thank you

**Comment by snikolayev** [comment, marked as answer]
@GianLucaBagnoli there are legit use cases where the same rule type may need to be instantiated multiple times. E.g. if the class is parameterized, and each instance is actually a different rule with different conditions. So, the RuleRepository cannot make assumptions about the rules' uniqueness solely based on the type. You are right - it is the caller's responsibility to ensure logical uniqueness of the rules in the repository.

**Reply by GianLucaBagnoli** [reply]
Ok. Thank you
